src/components/utils/src/timer.cc

```diff
@@ -64,6 +64,8 @@ timer::Timer::~Timer() {
   StopDelegate();
   single_shot_ = true;
 
+  thread_->set_delegate(NULL);
```

Comment:
@anosach-luxoft Checked - `ThreadDelegate` 's destructor sets `thread_ `delegate to `NULL`. Fixed in b1d62949f85705c676e053167f589ac9f1e982dd